src/gdgem/GdGemBase.cpp

```diff
@@ -403,74 +413,80 @@ void GdGemBase::processing_thread() {
+  Gem::TrackSerializer track_serializer(256, "nmx_tracks");
   track_serializer.set_callback(ProduceMonitor);
 
   HistogramSerializer hist_serializer(hists_.needed_buffer_size(), "nmx");
   hist_serializer.set_callback(ProduceMonitor);
 
-  Gem::TrackSerializer raw_serializer(1500, 1, "nmx_hits");
+  Gem::TrackSerializer raw_serializer(1500, "nmx_hits");
   raw_serializer.set_callback(ProduceHits);
 
   TSCTimer global_time, report_timer;
 
   unsigned int data_index;
+  int cnt = 0;
+  std::chrono::time_point<std::chrono::system_clock> timeEnd, timeStart;
+  timeStart = std::chrono::system_clock::now();
+  double avg = 0;
+  double total = 0;
+  int rep = 0;
   while (true) {
-    // mystats.fifo_free = input2proc_fifo.free();
+    // stats_.fifo_free = input2proc_fifo.free();
     if (!input2proc_fifo.pop(data_index)) {
-      mystats.processing_idle++;
+      stats_.processing_idle++;
       usleep(1);
     } else {
       auto len = eth_ringbuf->getDataLength(data_index);
       if (len == 0) {
-        mystats.fifo_seq_errors++;
+        stats_.fifo_seq_errors++;
       } else {
         builder_->process_buffer(
             eth_ringbuf->getDataBuffer(data_index), len);
-
-        // parser stats
-        mystats.fc_seq_errors = builder_->stats.parser_fc_seq_errors;
-        mystats.bad_frames = builder_->stats.parser_bad_frames;
-        mystats.good_frames = builder_->stats.parser_good_frames;
-        mystats.readouts_error_bytes = builder_->stats.parser_error_bytes;
-        mystats.readouts_total = builder_->stats.parser_readouts;
-
-        // builder stats
-        mystats.readouts_bad_geometry = builder_->stats.geom_errors;
-        mystats.readouts_bad_adc = builder_->stats.adc_rejects;
-        mystats.readouts_adc_zero = builder_->stats.adc_zero;
-        mystats.readouts_good += (builder_->hit_buffer_x.size()
+        cnt++;
+        if(cnt == 10000) {
+          cnt = 0;
+          timeEnd = std::chrono::system_clock::now();
+          int ms = std::chrono::duration_cast < std::chrono::milliseconds > (timeEnd - timeStart).count();
+          rep++;
+          total += ms*0.001;
+          avg = total/rep;
+          std::cout << rep << ": 10000 x process_buffer " << ms*0.001 << " s, avg = " << avg << " s\n";
+          timeStart = std::chrono::system_clock::now();
+        }
+        
+        if (nmx_opts.enable_data_processing) {
+          stats_.hits_good += (builder_->hit_buffer_x.size()
             + builder_->hit_buffer_y.size());
-
-        if (nmx_opts.send_raw_hits) {
-          Event dummy_event;
-          for (const auto& e : builder_->hit_buffer_x) {
-            dummy_event.ClusterA.insert(e);
-          }
-          for (const auto& e : builder_->hit_buffer_y) {
-            dummy_event.ClusterB.insert(e);
+          if (nmx_opts.send_raw_hits) {
+            Event dummy_event;
+            for (const auto& e : builder_->hit_buffer_x) {
+              dummy_event.ClusterA.insert(e);
+            }
```

Comment:
My first comment didn't make it for some reason

all the flag-like parameters like enable_data_processing, send_raw_hits etc
could (should?) be configurable from the command line. Having N json files with all the 
permutations or having to edit the json file every time you run a performance test is maybe not afficient?

Reply:
For the moment I'd at least like to keep the option to configure from the json file, since people know that there are all options together. But to add command line parameters is also a good idea.